src/sparseml/exporters/transforms/kv_cache/positions_adjustment_codegen.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from onnx import ModelProto, NodeProto
+
+from sparseml.exporters.transforms.kv_cache.positions_adjustment_base import (
+    PositionsAdjustmentBase,
+)
+from sparseml.exporters.transforms.utils.matching import get_structural_matches
+from sparseml.onnx.utils import ONNXGraph
+
+
+__all__ = ["PositionsAdjustmentCodeGen"]
+
+
+class PositionsAdjustmentCodeGen(PositionsAdjustmentBase):
```

Comment:
..... this is it? if a version upgrade simplifies what we need to do this significantly (both in terms of engineering hours as well as de-risking model/inference errors) definitely push for prioritize upgrade!

great work here

Reply:
I was totally overestimating the amount of work that is needed to do the transformer upgrade. Shame on me 😊